components/List/tests/unit/MDCSelfSizingStereoCellTests.m

```diff
@@ -0,0 +1,112 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MaterialList.h"
+
+#import <XCTest/XCTest.h>
+
+#import "MaterialTypography.h"
+
+@interface MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow : UIWindow
+
+/** Used to override the value of @c preferredContentSizeCategory. */
+@property(nonatomic, copy) UIContentSizeCategory contentSizeCategoryOverride;
+
+@end
+
+@implementation MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = UIContentSizeCategoryLarge;
+  }
+  return self;
+}
+
+- (instancetype)initWithContentSizeCategoryOverride:
+    (UIContentSizeCategory)contentSizeCategoryOverride {
+  self = [super init];
+  if (self) {
+    self.contentSizeCategoryOverride = contentSizeCategoryOverride;
+  }
+  return self;
+}
+
+- (UITraitCollection *)traitCollection {
+  if (@available(iOS 10.0, *)) {
+    UITraitCollection *traitCollection = [UITraitCollection
+        traitCollectionWithPreferredContentSizeCategory:self.contentSizeCategoryOverride];
+    return traitCollection;
+  }
+  return [super traitCollection];
+}
+
+@end
+
+@interface MDCSelfSizingStereoCellTests : XCTestCase
+
+@end
+
+@implementation MDCSelfSizingStereoCellTests
+
+- (void)testSelfSizingStereoCellDynamicTypeBehavior {
+  // Given
+  MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow *largeContainer =
+      [[MDCSelfSizingStereoCellTestsDynamicTypeContentSizeCategoryOverrideWindow alloc]
+          initWithContentSizeCategoryOverride:UIContentSizeCategoryLarge];
+  MDCSelfSizingStereoCell *cell = [[MDCSelfSizingStereoCell alloc] init];
+  [largeContainer addSubview:cell];
+  UIFont *titleFont = cell.titleLabel.font;
+  UIFont *detailFont = cell.detailLabel.font;
+  MDCFontScaler *headlineFontScaler =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleSubtitle1];
+  MDCFontScaler *detailFontScalar =
+      [[MDCFontScaler alloc] initForMaterialTextStyle:MDCTextStyleButton];
+  titleFont = [headlineFontScaler scaledFontWithFont:titleFont];
+  titleFont = [titleFont mdc_scaledFontAtDefaultSize];
+  detailFont = [detailFontScalar scaledFontWithFont:detailFont];
+  detailFont = [detailFont mdc_scaledFontAtDefaultSize];
+  cell.titleLabel.font = titleFont;
+  cell.detailLabel.font = detailFont;
```

Comment:
I noticed that when `detailFont` had the same `pointSize` as `cell.detailFont.font` this call to `-setFont:` didn't seem to do anything! And tests were failing. That's why I used `.button` instead of `.caption` because with `.caption`.

Reply:
Interesting. Was `detailFont` instance assigned to `cell.detailLabel.font`? This looks like something concerning for our Typography lib.

Reply:
I tested it locally. It happened, and I believe it is caused by UILabel's font setter.

Reply:
After this line:
`UIFont *detailFont = cell.detailLabel.font;`
`po cell.detailLabel.font` returned:
`<UICTFont: 0x7fb076a355c0> font-family: ".SFUIText"; font-weight: normal; font-style: normal; font-size: 12.00pt`

After this line:
`detailFont = [detailFont mdc_scaledFontAtDefaultSize];`
`po detailFont` returned:
`<UICTFont: 0x7fb076928260> font-family: ".SFUIText"; font-weight: normal; font-style: normal; font-size: 12.00pt`

After this line:
`cell.detailLabel.font = detailFont;`
`po cell.detailLabel.font` returned:
`<UICTFont: 0x7fb076a355c0> font-family: ".SFUIText"; font-weight: normal; font-style: normal; font-size: 12.00pt`

I'm guessing UILabel's `-setFont:` returns early if `[currentFont isEqual:newFont]` returns `YES` :/

Reply:
Let's add a TODO to change the Button style back to Caption after fixing this one.